src/Servers/Kestrel/shared/KnownHeaders.cs

```diff
@@ -843,7 +843,14 @@ internal partial class {loop.ClassName} : IHeaderDictionary
             }}
             set
             {{
-                {header.SetBit()};
+                if (!StringValues.IsNullOrEmpty(value))
```

Comment:
This is an existing bug in .NET 6 being fixed right?

Reply:
Should this particular change go into 6.0 RC2?

Reply:
Yes, and yes I plan to backport this entire PR to rc2.